prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -388,6 +449,28 @@ int rxr_pkt_init_eor(struct rxr_ep *ep, struct rxr_rx_entry *rx_entry, struct rx
 	return 0;
 }
 
+void rxr_pkt_handle_eor_send_completion(struct rxr_ep *ep,
+					struct rxr_pkt_entry *pkt_entry)
+{
+	struct rxr_eor_hdr *eor_hdr;
+	struct rxr_rx_entry *rx_entry;
+
+	eor_hdr = (struct rxr_eor_hdr *)pkt_entry->pkt;
+
+	rx_entry = ofi_bufpool_get_ibuf(ep->rx_entry_pool, eor_hdr->rx_id);
+	assert(rx_entry && rx_entry->rx_id == eor_hdr->rx_id);
+
+	if (rx_entry->op == ofi_op_msg || rx_entry->op == ofi_op_tagged) {
+		rxr_cq_write_rx_completion(ep, rx_entry);
+	} else {
+		assert(rx_entry->op == ofi_op_write);
+		if (rx_entry->cq_entry.flags & FI_REMOTE_CQ_DATA)
```

Comment:
Do we still need to write a truncated error when `FI_REMOTE_CQ_DATA` is not set?

Reply:
If FI_REMOTE_CQ_DATA is not set, we do not need to write completion, because  EFA does not support FI_RMA_EVENT.

Reply:
Right, I get that we do not need to write a completion in the normal case. But truncate is an exception, do we have to do anything special with that? Or just ignore it?

Reply:
I think we should check if target buffer length is smaller than source buffer length in the RMA requester, e.g. rxr_rma_readmsg() and rxr_rma_writemsg(), and return -FI_EINVAL if it is the case.

Reply:
Ah, yes that seems like the best thing to do.